team_builder_react/src/utilities/API_Interface.js

```diff
@@ -0,0 +1,233 @@
+                        lastName: '',
+                        skills: [], each element is { name: string, id: int }
+                        traits: [], each element is { name: string, score: int }
+                    }
+    
+    returns: if no server errors, a Promise that is the response from the server (https://axios-http.com/docs/res_schema).
+             if there was an error in the request, then the Axios error in a Promise.
+
+    What will actually be sent to the back-end:
+    {
+        "name_last": "Bar",
+        "name_first": "Foo",
+        "skills": "Python, Django",
+        "bpt_confidence": 0.1,
+        "bpt_delegator": 0.2,
+        "bpt_determination": 0.3,
+        "bpt_disruptor": 0.4,
+        "bpt_independence": 0.5,
+        "bpt_knowledge": 0.6,
+        "bpt_profitability": 0.7,
+        "bpt_relationship": 0.8,
+        "bpt_risk": 0.9,
+        "bpt_selling": 1.0
+    }
+*/
+export const SendPersonToBackEnd = async (person) => {
+    var objectToSend = {
+        name_last: person.lastName,
+        name_first: person.firstName,
+        skills: ""
+    };
+
+    // Add the skill name for each skill the user selected
+    objectToSend.skills = await CheckListArrayToString(person.skills);
+
+    person.traits.forEach(trait => {
+        objectToSend["bpt_" + trait.name] = trait.score;
+    })
+
+    try {
+        const response = await axiosInstance.post('/employees/', objectToSend);
+        return response;
+    } catch (error) {
+        return error;
+    }
+}
+
+/*
+    SendTeamParameters is an asynchronous function that takes the parameters by
+        which a new team will be built as parameters to the function. This
+        function will put the parameters into a JSON object and POST it to the
+        back-end so it can start the process of building a balanced team with
+        those parameters.
+     
+    parameters: teamSize - integer representing the number of team members the
+                    built team should have.
+
+                techSkills - an array of objects. Each element is { name: string, id: int }
+                    (as given by the CheckList component) and represents a
+                    technical skill that the build team should posses, if possible.
+
+    returns: if no server errors, a Promise that is the id of the newly created team
+             if there was an error in the request, -1 as a Promise and the error is printed to the console.
+
+    What will actually be sent to the back-end:
+    {
+        "teamSize": int
+        "skills": comma separated string of skills
+    }
+*/
+export const SendTeamParameters = async (teamSizeParam, techSkills) => {
+    var objectToSend = {
+        teamSize: teamSizeParam,
+        skills: CheckListArrayToString(techSkills)
+    };
+
+    try {
+        const response = await axiosInstance.post('/requested_teams/', objectToSend);
+        return response.data.id;
+    } catch (error) {
+        console.log(error);
```

Comment:
We should probably remove the console logs from production code.  Also, this function errors with a -1 instead of passing the caught error (like the other functions).  Is that purposeful or something left in?

Reply:
That is purposeful. I'm leaving it to the pages/components using these functions to handle the errors. Also, here the only thing the pages/components need from this response is the id of the team, so I return that. So that the function doesn't return an int sometimes and an error object other times, I used -1 as the error "flag."